embassy-boot/boot/src/lib.rs

```diff
@@ -619,12 +660,120 @@ impl FirmwareUpdater {
         }
     }
 
+    /// Verify the DFU given a public key. If there is an error then DO NOT
+    /// proceed with updating the firmware as it must be signed with a
+    /// corresponding private key (otherwise it could be malicious firmware).
+    ///
+    /// Mark to trigger firmware swap on next boot if verify suceeds.
+    ///
+    /// If the "ed25519-salty" feature is set (or another similar feature) then the signature is expected to have
+    /// been generated from a SHA-512 digest of the firmware bytes.
+    ///
+    /// If no signature feature is set then this method will always return a
+    /// signature error.
+    ///
+    /// # Safety
+    ///
+    /// The `_read_aligned` buffer must have a size of F::READ_SIZE, and follow the alignment rules for the flash being read from.
+    /// The `_write_aligned` buffer must have a size of F::WRITE_SIZE, and follow the alignment rules for the flash being written to.
+    #[cfg(feature = "_verify")]
+    pub async fn verify_and_mark_updated<F: AsyncNorFlash>(
+        &mut self,
+        _flash: &mut F,
+        _public_key: &[u8],
+        _signature: &[u8],
+        _update_len: usize,
+        _read_aligned: &mut [u8],
```

Comment:
Maybe instead of having to pass both a read_aligned and a write_aligned buffer, pass a single buffer that is write_aligned? Having to pass two buffers requires additional RAM for no reason.

I've not seen a case where write alignment didn't also cover read alignment for nrf, stm32 and rp, so it should be fine.

Reply:
`READ_SIZE` appears to be 1 on the nRF.